packages/api/test/functional/role.spec.js

```diff
@@ -0,0 +1,99 @@
+const { test, trait } = use('Test/Suite')('Role');
+trait('Test/ApiClient');
+trait('Auth/Client');
+trait('DatabaseTransactions');
+
+const Role = use('App/Models/Role');
+const User = use('App/Models/User');
+
+const role = {
+	role: 'TEST_ROLE',
+	description: 'Test Role',
+};
+
+const user = {
+	username: 'sabiatestinguser',
+	email: 'sabiatestingemail@gmail.com',
+	password: '123123',
+};
+
+test('try to access resource without authorization', async ({ client }) => {
+	await Role.create(role);
+
+	const response = await client.get('/roles').end();
+
+	response.assertStatus(401);
+});
+
+test('GET roles Show a list of all roles', async ({ client }) => {
+	await Role.create(role);
+
+	const loggeduser = await User.create(user);
+
+	const response = await client
+		.get('/roles')
+		.loginVia(loggeduser, 'jwt')
+		.end();
+
+	response.assertStatus(200);
+	response.assertJSONSubset([role]);
+});
+
+test('POST /roles create/save a new role.', async ({ client }) => {
+	const loggeduser = await User.create(user);
+
+	const response = await client
+		.post('/roles')
+		.send(role)
+		.loginVia(loggeduser, 'jwt')
+		.end();
+
+	response.assertStatus(200);
```

Comment:
Para que tenhamos certeza, faça uma consulta ao banco usando o model para ter certeza que role foi cadastrado no banco. Isso testa do início ao fim e faz com que não precisamos testar o model diretamente.

Caso tenha um bug que não esteja salvando no banco mas faça com que o controller retorna status 200 esse teste irá passar.